projects/igniteui-angular/src/lib/directives/text-selection/text-selection.directive.spec.ts

```diff
@@ -24,9 +24,10 @@ describe('IgxSelection', () => {
 
         const input = fix.debugElement.query(By.css('input')).nativeElement;
         input.focus();
-
-        expect(input.selectionEnd).toEqual(input.value.length);
-        expect(input.value.substring(input.selectionStart, input.selectionEnd)).toEqual(input.value);
+        requestAnimationFrame(() => {
```

Comment:
Ahem. This is a basic test, so nothing is waiting for the rAF callback and the expectations inside. One of those places where `fakeAsync` _can_ be useful